### ISSUE TITLE
Error when I use "Pick Color (left icon) and Pick brushstroke settings, stroke color, and layer (right icon)" on status bar.

### ISSUE THREAD

**Issue by americogobbo** [mypaint/mypaint#324]
#### Description
Bug Detected when you try to use the new icons features (Pick color and Pick brushstroke...) on status bar.

#### Steps to reproduce

Open MyPaint
Create a new document
Paint something or with empty canvas (the error is always present)
Pick on icons => "Pick color" or "Pick brushstroke settings..."

#### Traceback

```python
Mypaint version: 1.2.0-alpha.20150522+git.ea85620
System information: Linux-3.16.0-38-generic-x86_64-with-Ubuntu-14.04-trusty
Traceback (most recent call last):
  File "/home/jag/installs/mypaint/gui/picker.py", line 228, _start_grab(self=<gui.picker.ContextPickingGrabPresenter object>, device=<gtk.gdk.X11DeviceXI2 object at 0x7f1124455fa0 (GdkX11DeviceXI2 at 0x2f26220)>, time=7778791L, inibutton=None)
            assert (int(owner.get_events() & self._GRAB_MASK) == int(self._GRAB_MASK)), \
                "Grab owner's events must match %r" % (self._GRAB_MASK,)
  variables: {'self._GRAB_MASK': ('local', <flags GDK_BUTTON_MOTION_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK of type GdkEventMask>)}
AssertionError: Grab owner's events must match <flags GDK_BUTTON_MOTION_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK of type GdkEventMask>
```

**Comment by achadwick** [comment]
Okay, that's interesting. What DE are you running please?

Can you patch with

```diff
diff --git a/gui/picker.py b/gui/picker.py
index 2313795..cbf0981 100644
--- a/gui/picker.py
+++ b/gui/picker.py
@@ -221,9 +221,7 @@ class PickingGrabPresenter (object):
         assert window is not None
 
         # Ensure that it'll receive termination events.
-        # Perhaps could just do
-        #     owner.add_events(self._GRAB_MASK)
-        # here? It may be unnecessary though. Um.
+        owner.add_events(self._GRAB_MASK)
         assert (int(owner.get_events() & self._GRAB_MASK) == int(self._GRAB_MASK)), \
             "Grab owner's events must match %r" % (self._GRAB_MASK,)
````

temporarily to see if it makes the issue go away? Thanks.

**Comment by achadwick** [comment]
@americogobbo Please try the patch above, if you can.

What desktop environment are you running here?
Any special hardware?
What OS and version is this?

**Comment by americogobbo** [comment]
My DE is GNOME 3.10 Ubuntu Gnome 14.04.2 LTS
My box is an Intel® Core™ i5-3570K CPU @ 3.40GHz × 4; 8Gb Memory; Graphics - Gallium 0.4 on AMD CAICOS (opensource driver)

**Comment by Ico-dY** [comment]
Same problem, here:
```python
Mypaint version: 1.2.0-alpha.20150604+git.266e56e
System information: Linux-3.13.0-53-generic-x86_64-with-Ubuntu-14.04-trusty
Traceback (most recent call last):
  File "/usr/local/share/mypaint/gui/picker.py", line 228, _start_grab(self=<gui.picker.ContextPickingGrabPresenter object>, device=<gtk.gdk.X11DeviceXI2 object at 0x7fbed479bbe0 (GdkX11DeviceXI2 at 0x1833210)>, time=50121575L, inibutton=None)
            assert (int(owner.get_events() & self._GRAB_MASK) == int(self._GRAB_MASK)), \
                "Grab owner's events must match %r" % (self._GRAB_MASK,)
  variables: {'self._GRAB_MASK': ('local', <flags GDK_BUTTON_MOTION_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK of type GdkEventMask>)}
AssertionError: Grab owner's events must match <flags GDK_BUTTON_MOTION_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK of type GdkEventMask>
```

Just tried the patch and it works for me! ^^
Xubuntu 14.04.2 LTS + Xfce 4.10; Intel® Core™ i7-4790K + 32Gb Memory; NVidia closed source driver v346.72

**Comment by achadwick** [comment]
Reproduced with Xubuntu 14.04, using vboxvideo_drv.so for xorg 1.15.1, module version 1.0.1. I'm guessing it's an oddity with either libgtk-3-0 version 3.10.8-0ubuntu1.4 or that version of xorg.
